oida/commands/componentize.py

```diff
@@ -185,3 +196,140 @@ class {class_name}(AppConfig):
+        if self.in_app > 0:
+            # Test if the name of the task is not explicitly set
+            if not m.matches(
+                original_node,
+                m.Call(
+                    args=[m.ZeroOrMore(), m.Arg(keyword=m.Name("name")), m.ZeroOrMore()]
+                ),
+            ):
+                arguments = original_node.args
+                # The implicit name of the task is the path to the old module concatenated with the function name.
+                task_name = f'"{self.old_module}.{self.function_name}"'
+                arguments = (  # type: ignore
+                    cst.Arg(
+                        keyword=cst.Name("name"),
+                        value=cst.SimpleString(value=task_name),
+                        # Equal and comma are not strictly necessary - the code would be functional without these.
+                        # They are needed however to comply with coding conventions and pass the unit test.
+                        equal=cst.AssignEqual(
+                            whitespace_before=cst.SimpleWhitespace(value=""),
+                            whitespace_after=cst.SimpleWhitespace(value=""),
+                        ),
+                        comma=cst.Comma(
+                            whitespace_before=cst.SimpleWhitespace(value=""),
+                            whitespace_after=cst.ParenthesizedWhitespace(
+                                first_line=cst.TrailingWhitespace(
+                                    whitespace=cst.SimpleWhitespace(value=""),
+                                    comment=None,
+                                    newline=cst.Newline(value=None),
+                                ),
+                                empty_lines=[],
+                                indent=True,
+                                last_line=cst.SimpleWhitespace(value="    "),
+                            ),
+                        ),
+                    ),
+                ) + arguments
+                new_node = updated_node.with_changes(args=arguments)
+                return new_node
+        return updated_node
+
+    def visit_FunctionDef(self, node: cst.FunctionDef) -> Optional[bool]:
+        if m.matches(
+            node,
+            m.FunctionDef(
+                decorators=[
+                    m.ZeroOrMore(),
+                    m.Decorator(
+                        m.Call(m.Attribute(value=m.Name("app"), attr=m.Name("task")))
+                    ),
+                    m.ZeroOrMore(),
+                ]
+            ),
+        ):
+            # We need the name of the function to recreate the implicit task name.
+            self.function_name = node.name.value
+        return super().visit_FunctionDef(node)
+
+    def leave_FunctionDef(
+        self, original_node: cst.FunctionDef, updated_node: cst.FunctionDef
+    ) -> Union[BaseStatement, FlattenSentinel[BaseStatement], RemovalSentinel]:
+        self.function_name = None
+        return updated_node
```

Comment:
I'm not fully following why you've split this up and keep state, wouldn't it be much easier to just do everything here?

Ie. something like this:

```python
def update_decorator(self, decorator: Decorator) -> Decorator:
    # If name not in arguments add it

def leave_FunctionDef(self, ...) -> ...:
    celery_task_decorator = m.Decorator(...)
    decorators = [
        self.update_decorator(decorator) if m.matches(celery_task_decorator, decorator) else decorator
        for decorator in updated_node.decorators 
    ]

    return updated_node.with_changes(decorators=decorators)
```

Reply:
Done. The number of lines of code  has decreased quite a lot. Thanks for the suggestion.